cmd/fs-v1.go

```diff
@@ -452,13 +452,17 @@ func (fs fsObjects) GetObject(bucket, object string, offset int64, length int64,
 // getObjectInfo - wrapper for reading object metadata and constructs ObjectInfo.
 func (fs fsObjects) getObjectInfo(bucket, object string) (oi ObjectInfo, e error) {
 	fsMeta := fsMetaV1{}
-	if hasSuffix(object, slashSeparator) {
-		// Directory call needs to arrive with object ending with "/".
-		fi, err := fsStatDir(pathJoin(fs.fsPath, bucket, object))
-		if err != nil {
-			return oi, toObjectErr(err, bucket, object)
+	fi, err := fsStatDir(pathJoin(fs.fsPath, bucket, object))
+	if err != nil && errorCause(err) != errFileAccessDenied {
+		return oi, toObjectErr(err, bucket, object)
+	}
+	if fi != nil {
```

Comment:
Can you add a test case for this? @nitisht

Reply:
sure will do.